src/coredump/coredump.c

```diff
@@ -1115,18 +1115,36 @@ static int process_socket(int fd) {
                                 /* We have all FDs we need let's take a shortcut here. */
                                 break;
                         } else {
+                                struct cmsghdr *cmsg;
+                                unsigned n_fds = 0;
+
                                 found = cmsg_find(&mh, SOL_SOCKET, SCM_RIGHTS, CMSG_LEN(sizeof(int)));
-                                if (found)
+                                if (first) {
+                                        /* This is the first message that carries file descriptors, maybe there will be
+                                         * one more that actually contains array of two descriptors. */
+                                        assert(input_fd < 0);
+
                                         input_fd = *CMSG_TYPED_DATA(found, int);
-                        }
+                                        first = false;
 
-                         /* This is the first message that carries file descriptors, maybe there will be one more that actually contains array of descriptors. */
-                        if (first) {
-                                first = false;
-                                continue;
-                        }
+                                        continue;
+                                }
 
-                        break;
+                                /* This is second iteration and we didn't find array of two FDs, hence we either
+                                 * have no FDs which is OK and we can break or we have some other number of FDs
+                                 * and somebody is playing games with us. So let's check for that. */
+                                CMSG_FOREACH(cmsg, &mh)
+                                        if (cmsg->cmsg_level == SOL_SOCKET && cmsg->cmsg_type == SCM_RIGHTS)
+                                                n_fds++;
```

Comment:
you can use cmsg_find() and specify the size as (socklen_t) -1

Reply:
Hmmm, I want to count number of FDs attached to the message how does cmsg_find() helps me in that ?

Reply:
SCM_RIGHTS should not be included multiple times.

Reply:
Good point, so now I see what Lennart meant.